packages/salesforcedx-vscode-core/src/index.ts

```diff
@@ -395,6 +395,15 @@ export async function activate(context: vscode.ExtensionContext) {
     sfdxProjectOpened
   );
 
+  // Set internal dev context
+  const internalDev = sfdxCoreSettings.getInternalDev();
```

Comment:
The piece of code that will populate a global context according to whether it finds the new setting `salesforcedx-vscode-core.internal-development` and its value .